### DISCUSSION TITLE
French Translation of lessons.

### DISCUSSION THREAD

**Discussion by ml4nC3** [GDQuest/learn-gdscript#790] [Ideas]
Hi,
I'm a programming teacher in France, and started a small workshop about making games with a few motivated students. Some of them asked to work on Godot and I liked it because I wanted to go deeper in understanding how it works.

I liked a lot the thoughts and ideas that you shared about learning programming, and I wondered if I could give a hand on this learning tool by starting a french translation of the lessons.

I'm no translator but I think I'm skilled enough in english to translate this in french considering I know the technical vocabulary.

How hard would it be ? I can see there is already a Spanish version so I assume there is already a mechanism to add new languages.

Let me know how you feel about this.

**Comment by NathanLovato** [comment]
There were already complete French translations for a previous version. You can find all translation files here: https://github.com/GDQuest/learn-gdscript-translations

We're missing some CI or script to check translations are up to date and automatically integrate or remove languages into releases depending on how complete the translations are. Also, we would need such a tool to warn us about translations that became stale.

We'd like to get to it someday, but it's a very low priority now. We have many higher-priority tasks to attend to.

If you or someone you know would like to contribute this kind of code, that'd be most welcome. Translations to French specifically, we have a professional translator available to work on this, the person who took care of the original translations to Spanish and French, so for these two languages specifically we'd favor working with her.

**Reply by ml4nC3** [reply]
It makes sense of course. Better to work with a professional.  
I can see in translation files that each paragraph is described with it's english text as some sort of ID and the translated version. Also there is those kind of paragraph ids like "#: course/lesson-10-the-game-loop/lesson.tres:14"

The script you're talking about should work on those files ? I guess there is a reference somewhere with english paragraphs, and you expect this script to go through translation files and identifies english paragraphs that does not appear in the translation files. Is that correct ?

What language should be used ? Is there somewhere, some documentation that could give needed information ? I might be able to help on that.

**Reply by NathanLovato** [reply]
Yes, we need code that would check these .POT and .PO files and the app's source repository. 

For the language it could be python (ideal), bash (second best), or perhaps javascript - basically something anyone in our team can easily maintain. I can write a comprehensive issue about this after the Godot 4.0 release rush is over with the requirements. There are programs that exist and can possibly help (`gettext` package on Linux, also `po4all`).

Is it fine by you if I write a detailed issue... let's say by next week? I'll write it down but don't hesitate to ping me if I forget, I've been in crunch almost all of February for the Godot 4 release, and we're moving to preparing for the game developer conference in 3 weeks.

**Reply by ml4nC3** [reply]
Python is great, it is the language on which I have the more knowledge.

Don't worry, no rush. I have some work to do on my side too so I won't bore until you get the chance to create this issue ! ;)

I'll get back to you if I have no news within a week or two. Good luck for the next release and thanks for your work !

**Reply by ml4nC3** [reply]
Hello NathanLovato,
How is going the Godot 4 event ?
I'm still opened for contribution on that matter if you get the chance to write a detailed ticket. Let me know !

**Reply by NathanLovato** [reply]
Absolutely, sorry for the delay, March ended up being just as busy as February unfortunately. It's on my todolist for the next few days

**Reply by NathanLovato** [reply]
There you go! Thanks for your patience and the reminder: https://github.com/GDQuest/learn-gdscript/issues/813
